content/docs/userspace/dist/guide.md

```diff
@@ -0,0 +1,281 @@
++++
+title = "Guide"
+weight = 4
+template = "doc.html"
++++
+
+In this document we'll walk through an example of creating and publishing a desk that others can install. We'll create a simple "Hello World!" front-end with a "Hello" tile to launch it. For simplicity, the desk won't include an actual Gall agent, but we'll note everything necessary if there were one.
+
+## Create desk
+
+To begin, we'll need to clone the [urbit git repo](https://github.com/urbit/urbit) from the Unix terminal:
+
+```
+[user@host ~]$ git clone https://github.com/urbit/urbit urbit-git
+```
+
+Once that's done, we can navigate to the `pkg` directory in our cloned repo:
+
+```
+[user@host ~]$ cd urbit-git/pkg
+[user@host pkg]$ ls .
+arvo      btc-wallet    garden        grid  interface   npm                webterm
+base-dev  docker-image  garden-dev    herb  landscape   symbolic-merge.sh
+bitcoin   ent           ge-additions  hs    libaes_siv  urbit
+```
+
+Each desk defines its own `mark`s, in its `/mar` folder. There are no longer shared system marks that all userspace code knows, nor common libraries in `/lib` or `/sur`. Each desk is completely self-contained. This means any new desk will need a number of base files.
+
+To make the creation of a new desk easier, `base-dev` and `garden-dev` contain symlinks to all `/sur`, `/lib` and `mark` files necessary for interacting with the `%base` and `%garden` desks respectively. These dev desks can be copied and merged with the `symbolic-merge.sh` included.
+
+Let's create a new `hello` desk:
+
+```
+[user@host pkg]$ ./symbolic-merge.sh base-dev hello
+[user@host pkg]$ ./symbolic-merge.sh garden-dev hello
+[user@host pkg]$ cd hello
+[user@host hello]$ ls
+lib  mar  sur
+```
+
+### `sys.kelvin`
+
+Our desk must include a `sys.kelvin` file which specifies the kernel version it's compatible with. Let's create that:
+
+```
+[user@host hello]$ echo "[%zuse 420]" > sys.kelvin
+[user@host hello]$ cat sys.kelvin
+[%zuse 420]
+```
+
+### `desk.ship`
+
+We can also add a `desk.ship` file to specify the original publisher of this desk. We'll try this on a fakezod so let's just add `~zod` as the publisher:
+
+```
+[user@host hello]$ echo "~zod" > desk.ship
+[user@host hello]$ cat desk.ship
+~zod
+```
+
+### `desk.bill`
+
+If we had Gall agents in this desk which should be automatically started when the desk is installed, we'd add them to a `hoon` list in the `desk.bill` file. It would look something like this:
+
+```hoon
+:~  %some-app
+    %another
+==
+```
+
+In this example we're not adding any agents, so we'll simply omit the `desk.bill` file.
+
+### `desk.docket-0`
+
+The final file we need is `desk.docket-0`. This one's more complicated, so we'll open it in our preferred text editor:
+
+```
+[user@host hello]$ vim desk.docket-0
```

Comment:
Consider, for comedic effect:

```suggestion
[user@host hello]$ ed desk.docket-0
```

Alternatively, if we're concerned about unwitting users actually running this, probably `nano` instead, because it has on-screen usage docs?